lib/steps.js

```diff
@@ -619,6 +619,83 @@ Then('I sign', async () => {
 
 
 
+//#####################################
+//#####################################
+// Macros
+//#####################################
+//#####################################
+
+When(/I set the name of ?(?:the var(?:iable)?)? "([^"]+)" to "([^"]+)"/, async (var_base, name_str) => {
+  /* Given a variable name and a space-separated string, fills the
+  *    appropriate fields on an Assembly Line name question.
+  *    It handles any of:
+  *    1. A one-word value to set name.text (ex: 'LandlordEtAl')
+  *    1. An Individual's name that it will set based on the number of words:
+  *       1. Two words - a first and last name (ex: "Tal Taylor")
+  *       1. Three words - a first, middle, and last name (ex: "Tal Tay Taylor")
+  *       1. Four words - a first, middle, last, and suffix (ex: "Tal Tay Taylor III")
+  * 
+  * The string 'the var'/'the variable' part of the step is optional, as you can see below.
+  * 
+  * Examples:
+  * 1. I set the name of "user" to "Uli Uster"
+  * 1. I set the name of the var "defendant" to "Dalia Day Defo Jr"
+  * 1. I set the name of the variable "defendant" to "Dalia Day Defo Jr"
+  */
+  let name_patterns = [
+    [ `text` ],
+    [ `first`, `last` ],
+    [ `first`, `middle`, `last` ],
+    [ `first`, `middle`, `last`, `suffix` ],
+  ];
+
+  let name_parts = name_str.split(' ');
+  let desired_pattern = name_patterns[( name_parts.length - 1 )];
+
+  for ( let parti = 0; parti < name_parts.length; parti++ ) {
+    let var_name = `${ var_base }.name.${ desired_pattern[ parti ] }`
+    let value = name_parts[ parti ];
+    await scope.setVar( scope, var_name, value );
+  }
+
+  // // Continue to next page...?
+  // await scope.continue( scope );
+  await scope.afterStep(scope, {waitForShowIf: true});
+});
+
+
+When(/I set the address of ?(?:the var(?:iable)?)? "([^"]+)" to "([^"]+)"/, async (var_base, address_str) => {
+  /* Given a variable name and a four-part address formatted as below,
+  *    sets the appropriate Assembly Line address fields.
+  * 
+  * The string 'the var'/'the variable' part of the step is optional, as you can see below.
+  * 
+  * Examples:
+  * 1. When I set the address of "users[0]" to "112 Southampton St., Unit 1, Boston, MA 02118"
+  * 1. When I set the address of the variable "users[0]" to "112 Southampton St., Unit 1, Boston, MA 02118"
+  * 1. When I set the address of the var "users[0]" to "112 Southampton St., Unit 1, Boston, MA 02118"
+  */
+  // Get all the parts of the address properly separated
+  let split = address_str.split(/,\s*/);
+  let last_parts = (split.pop()).split(/\s+/);
+  let address_parts = split.concat( last_parts );
```

Comment:
some clarifying comments here would be helpful to understand the output at each stage